site/content/extend/plugins/best-practices.md

```diff
@@ -6,6 +6,67 @@ weight: 50
 
 See here for [server-specific best practices for plugins](/extend/plugins/server/best-practices/). Webapp-specific best practices are incoming.
 
+## How can I create my own plugin setting?
+
+Plugins can define their admin settings in the plugin's manifest [setting schema](https://developers.mattermost.com/extend/plugins/manifest-reference/#settings_schema) which include a pre-defined types that the System Console UI currently supports. 
```

Comment:
I wonder if we should expand this section in order to emphasize that this is the "normal" way of configuring custom plugins and avoid someone diving into a custom component prematurely. Maybe even break it out as two questions, "How do I allow my plugin to be configured?" and "How do I handle more complex plugin configurations?"

Reply:
@lieut-data are you suggesting adding a new question before 
"How do I allow my plugin to be configured?" and change this new content to "How do I handle more complex plugin configurations?"

Reply:
@lieut-data should we create an improvement ticket for this suggestion ?

Reply:
Ticket created - [MM-21364](https://mattermost.atlassian.net/browse/MM-21364)